dataset/src/test/java/org/keycloak/benchmark/it/DeploymentIT.java

```diff
@@ -0,0 +1,278 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.keycloak.benchmark.it;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.io.FileUtils;
+import org.awaitility.Awaitility;
+import org.junit.Assert;
+import org.junit.Assume;
+import org.junit.Before;
+import org.junit.Test;
+import org.keycloak.benchmark.dataset.TaskResponse;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * This tests the deployment of the dataset provider in an integration test.
+ * To do this, it expects Keycloak to be extracted in the target folder by the maven-dependency-plugin.
+ *
+ * @author Alexander Schwartz
+ */
+
+public class DeploymentIT {
+
+    private int port;
+    private String keycloak;
+
+    /**
+     * This can be run from the dataset module folder from the IDE for simple testing.
+     */
+    public static void main(String[] args) throws IOException, URISyntaxException, ExecutionException, InterruptedException {
+        packageProvider();
+        new DeploymentIT().startService(args);
+    }
+
+    public DeploymentIT() {
+        port = 8080;
+        keycloak = "http://127.0.0.1:" + port + "/";
+    }
+
+    @Before
+    public void setup() throws IOException {
+        // use a random free port to allow running the tests while other instances of Keycloak run at the same time
+        ServerSocket s = new ServerSocket(0);
+        port = s.getLocalPort();
+        s.close();
+        keycloak = "http://127.0.0.1:" + port + "/";
+    }
+
+    @Test
+    public void deployWithLegacyStore() throws IOException, ExecutionException, InterruptedException, URISyntaxException {
+        runTestWithParameter();
+    }
+
+    @Test
+    public void deployWithMapStore() throws IOException, ExecutionException, InterruptedException, URISyntaxException {
+        runTestWithParameter("--storage", "chm");
```

Comment:
@ahus1 does this take values such as ```runTestWithParameter("--storage", "hotrod")``` ? If I create a new test with this option and run, it currently fails